requests/models.py

```diff
@@ -403,11 +403,13 @@ def prepare_url(self, url, params):
     def prepare_headers(self, headers):
         """Prepares the given HTTP headers."""
 
+        self.headers = CaseInsensitiveDict()
         if headers:
-            valid_headers = filter(is_valid_header, headers.items())
-            self.headers = CaseInsensitiveDict((to_native_string(name), val) for name, val in valid_headers)
-        else:
-            self.headers = CaseInsensitiveDict()
+            for header in headers.items():
+                # Raise exception on invalid header value.
+                is_valid_header(header)
```

Comment:
This seems slightly kludgy to me, since we're not doing a transform or using this as a conditional. Maybe it's just a naming issue though. Renaming it `check_header_validity` and removing the `return True` may be a better option to describe it's current functionality. Otherwise, I'll leave it in, as is, to use as a generalized utility check elsewhere if needed.